File: 523. Continuous Subarray Sum/C++/Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int> &nums, int k) {
        unordered_map<int, int> memory;
        memory[0] = -1;
        int currentSum = 0;
        for (int i = 0; i < nums.size(); i++) {
            currentSum += nums[i];
            if (k != 0) {
                currentSum %= k;
            }
            if (memory.find(currentSum) != memory.end()) {
                if (i - memory[currentSum] > 1) {
                    return true;
                }
            } else {
                memory[currentSum] = i;
            }
        }
        return false;
    }
};

File: 523. Continuous Subarray Sum/Java/Continuous Subarray Sum.java
public class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> memory = new HashMap<>();
        memory.put(0, -1);
        int currentSum = 0;
        for (int i = 0; i < nums.length; i++) {
            currentSum += nums[i];
            if (k != 0) {
                currentSum %= k;
            }
            if (memory.containsKey(currentSum)) {
                if (i - memory.get(currentSum) > 1) {
                    return true;
                }
            } else {
                memory.put(currentSum, i);
            }
        }
        return false;
    }
}